pkg/operator/operator.go

```diff
@@ -73,7 +78,60 @@ type Operator struct {
 
 // New creates (but does not start) a new operator from configuration.
 func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, error) {
+	ctx := context.Background()
+	ctx, cancelFn := context.WithCancel(ctx)
+	go func() {
+		defer cancelFn()
+		<-config.Stop
+	}()
+
 	scheme := operatorclient.GetScheme()
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kube client: %w", err)
+	}
+	namespaceInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 24*time.Hour, informers.WithNamespace(operatorcontroller.DefaultOperandNamespace))
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(config.Namespace), "cluster-ingress-operator", &corev1.ObjectReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Namespace:  config.Namespace,
+		Name:       "ingress-operator",
+	})
+
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	desiredVersion := config.OperatorReleaseVersion
+	missingVersion := statuscontroller.UnknownVersionValue
+
+	// by default, this will exit(0) the process if the featuregates ever change to a different set of values.
```

Comment:
I usually ask people to write full sentences in comments.
```suggestion
	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
```